### ISSUE TITLE
TODO: write a script to provision EC2 agents; add throttling

### ISSUE THREAD

**Issue by hcho3** [hcho3/xgboost-devops#2]
For now, just run the script inside the Jenkins master. Lambda functions may be better but is more difficult to develop and test.

Goal: retire EC2 plugin in Jenkins. We should have full control over how EC2 agents are provisioned.

**Comment by chyunsu3** [comment]
Also: add throttling to meet daily/monthly spending limit. No EC2 workers should be provisioned when the spending limit is breached.

**Comment by chyunsu3** [comment]
I decided against writing a custom script for provisioning EC2 agents. [The MXNet project has something similar](https://github.com/apache/incubator-mxnet-ci/tree/master/services/jenkins-autoscaling/lambda_mxnet_ci/autoscaling) and the logic is quite complex and will likely require lots of maintenance down the road.

So I decided on a much simpler approach:
* Keep using the EC2 plugin in Jenkins, and let Jenkins handle provisioning EC2 agents.
* Write a small AWS Lambda function to enforce a daily budget. This is the Cost Watcher.
* Each testing job should invoke the Lambda function to ensure the budget is not exceeded (dmlc/xgboost#5884). If the budget is exceeded, test jobs are interrupted.
